fw/cache.c

```diff
@@ -329,18 +330,55 @@ tfw_cache_key_node(unsigned long key)
 }
 
 /**
- * Just choose any CPU for each node to use queue_work_on() for
- * nodes scheduling. Reserve 0th CPU for other tasks.
+ * Release node-cpu map.
  */
 static void
+tfw_release_node_cpus(void)
+{
+	int node;
+
+	for(node = 0; node < nr_online_nodes; node++) {
+		if(c_nodes[node].cpu)
```

Comment:
You doing:
`c_nodes = kzalloc(nr_online_nodes * sizeof(CaNode), GFP_KERNEL);` and in case of allocation error you invoking `tfw_release_node_cpus()`, it implies you must check `c_nodes` here, otherwise you will got null ptr dereference. Also you can use `kfree(c_nodes[node].cpu);` without pre-check as @keshonok mentioned. I don't see any places where garbage can be moved to `c_nodes`.